paddle/fluid/operators/log_softmax_op.cc

```diff
@@ -31,9 +31,17 @@ class LogSoftmaxOp : public framework::OperatorWithKernel {
  protected:
   framework::OpKernelType GetExpectedKernelType(
       const framework::ExecutionContext& ctx) const override {
-    return framework::OpKernelType(
-        OperatorWithKernel::IndicateVarDataType(ctx, "X"),
-        ctx.device_context());
+    auto input_data_type =
+        framework::OperatorWithKernel::IndicateVarDataType(ctx, "X");
+
+#ifdef PADDLE_WITH_MKLDNN
+    if (this->CanMKLDNNBeUsed(ctx, input_data_type)) {
+      return framework::OpKernelType(input_data_type, ctx.GetPlace(),
+                                     framework::DataLayout::kMKLDNN,
+                                     framework::LibraryType::kMKLDNN);
+    }
+#endif
+    return framework::OpKernelType(input_data_type, ctx.GetPlace());
```

Comment:
The method used to return framework::OpKernelType(input_data_type, ctx.device_context()) when mkldnn was not used, but now returns framework::OpKernelType(input_data_type, ctx.GetPlace()). Is there a difference between ctx.device_context() and ctx.GetPlace()?

Reply:
I'm the ghost :)

Reply:
@piotrekobi yes, there is a difference between `ctx.device_context()` and `ctx.GetPlace()`, but in this particular case `OpKernelType`'s constructor is overloaded and we achieve identical result whether we pass `ctx.device_context()` or `ctx.GetPlace()`. Code that is responsible for that is linked below
https://github.com/PaddlePaddle/Paddle/blob/058e1d8592e44fc913365520455d124333224adf/paddle/fluid/framework/operator.h#L389

https://github.com/PaddlePaddle/Paddle/blob/058e1d8592e44fc913365520455d124333224adf/paddle/fluid/framework/op_kernel_type.h#L39-L58